### ISSUE TITLE
Training on a single GPU - setting learning rate

### ISSUE THREAD

**Issue by PedroCHS** [Westlake-AI/A2MIM#4]
Thank you for your brilliant work.

I would like to apply A2MIM to pre-train an encoder on a custom dataset. I managed to run the code using a batch size of 128 (instead of 256, due to memory constraints) on a single GPU. I would like to ask for your help on how to set the learning rate in this case. 

In the "r50_l3_sz224_init_8xb256_cos_epX00.py" config file, it says that:
lr=3e-4 * 2048 / 512,  # 1.2e-3 for bs2048

Thanks in advance for your help. It will be highly appreciated.

**Comment by Lupin1998** [comment]
Hi, @PedroCHS. Sorry for the late reply. According to the linear scaling law of the learning rate, we should make sure the whole batch size (i.e., 2048) is equal to "batch size per GPU" (imgs_per_gpu) $\times$ "GPU number" $\times$ "gradient accumulation" (update_interval). Taking the config file you mentioned as an example, you should set `update_interval` to 16 (i.e., 2048 / 128) with a single GPU and a batch size of 128.

**Comment by Lupin1998** [comment]
If you have no more questions, I will close this issue. Feel free to open a new issue if more question occurs.
